src/platform/vita/vita_input.c

```diff
@@ -0,0 +1,428 @@
+            event.motion.xrel = xrel;
+            event.motion.yrel = yrel;
+            SDL_PushEvent(&event);
+        }
+    }
+
+    // map right stick to cursor keys
+    float right_x = SDL_JoystickGetAxis(joy, 2);
+    float right_y = -1 * SDL_JoystickGetAxis(joy, 3);
+    float right_joy_dead_zone_squared = 10240.0*10240.0;
+    float slope = 0.414214f; // tangent of 22.5 degrees for size of angular zones
+
+    if ((right_x * right_x + right_y * right_y) > right_joy_dead_zone_squared)
+    {
+        int up = 0;
+        int down = 0;
+        int left = 0;
+        int right = 0;
+
+        // upper right quadrant
+        if (right_y > 0 && right_x > 0)
+        {
+            if (right_y > slope * right_x)
+                up = 1;
+            if (right_x > slope * right_y)
+                right = 1;
+        }
+        // upper left quadrant
+        else if (right_y > 0 && right_x <= 0)
+        {
+            if (right_y > slope * (-right_x))
+                up = 1;
+            if ((-right_x) > slope * right_y)
+                left = 1;
+        }
+        // lower right quadrant
+        else if (right_y <= 0 && right_x > 0)
+        {
+            if ((-right_y) > slope * right_x)
+                down = 1;
+            if (right_x > slope * (-right_y))
+                right = 1;
+        }
+        // lower left quadrant
+        else if (right_y <= 0 && right_x <= 0)
+        {
+            if ((-right_y) > slope * (-right_x))
+                down = 1;
+            if ((-right_x) > slope * (-right_y))
+                left = 1;
+        }
+
+        if (!up_pressed && up) {
+            SDL_Event event;
+            event.type = SDL_KEYDOWN;
+            event.key.keysym.sym = SDLK_UP;
+            SDL_PushEvent(&event);
+        } else if (!down_pressed && down) {
+            SDL_Event event;
+            event.type = SDL_KEYDOWN;
+            event.key.keysym.sym = SDLK_DOWN;
+            SDL_PushEvent(&event);
+        }
+        if (!left_pressed && left) {
+            SDL_Event event;
+            event.type = SDL_KEYDOWN;
+            event.key.keysym.sym = SDLK_LEFT;
+            SDL_PushEvent(&event);
+        } else if (!right_pressed && right) {
+            SDL_Event event;
+            event.type = SDL_KEYDOWN;
+            event.key.keysym.sym = SDLK_RIGHT;
+            SDL_PushEvent(&event);
+        }
+    }
+}
+
+void vita_handle_virtual_keyboard()
+{
+    if (vkbd_requested) {
+        vkbd_requested = 0;
+        vita_start_text_input("", 0);
+    }
+}
+
+static void vita_start_text_input(char *initial_text, int multiline)
+{
+    char *text = vita_keyboard_get("Enter New Text:", initial_text, 600, multiline);
+    if (text != NULL)  {
+        for (int i = 0; i < 600; i++) {
+            SDL_Event down_event;
+            down_event.type = SDL_KEYDOWN;
+            down_event.key.keysym.sym = SDLK_BACKSPACE;
+            down_event.key.keysym.mod = 0;
+            SDL_PushEvent(&down_event);
```

Comment:
Put this in a helper function.

Reply:
Fixed in 35f42ab00e2b28a49e966a752eb3016a334d53bb